plugins/plugin_utils/keys_filter.py

```diff
@@ -37,31 +37,31 @@ def _keys_filter_params(data, target, matching_parameter):
             msg = "Top level keys must be strings. keys: %s"
             raise AnsibleFilterError(msg % elem.keys())
 
-    if not isinstance(target, Sequence):
-        msg = ("The target must be a string or a list. target is %s.")
-        raise AnsibleFilterError(msg % target)
```

Comment:
If you remove this check, the next `if` can fail if `target` does not allow `len()` to be applied.

I guess you should also move the next `if` into the other functions.

Reply:
Done. Implemented in the last PR.